src/openrct2/ride/RideRatings.cpp

```diff
@@ -2268,4 +2268,6 @@ bool RatingTuple::isNull() const
 void RatingTuple::setNull()
 {
     excitement = kRideRatingUndefined;
+    intensity = kRideRatingUndefined;
+    nausea = kRideRatingUndefined;
```

Comment:
This does not make any sense and also breaks the game a bit, kRideRatingUndefined is not meant to be used for anything other than excitement as it acts as a value for marking it as invalid/null, null is perhaps not the best choice here. As for the sort problem, perhaps the best choice is to fix the sort order and not modify random variables.

Reply:
Fixed the sort order - checking for ratings.isNull() on both sides before comparing between them.
Checking for == NULL for nausea and intensity directly wouldn't work, since if you are retesting a ride, it retains those ratings (hence the rides staying in same place on the sorted list before the fix) and excitement is set to the kRideRatingUndefined.
So all gameplay should now be as intended, only difference is untested rides appearing at the tail of the sorted rating lists always.

Reply:
Just seeing this now, will be more careful with future commits.

Reply:
> Just seeing this now, will be more careful with future commits.

Don't worry I think it was easy mistake to make, since 1/3 ratings worked the way you wanted the other 2 to work.
If the kRideRatingUndefined was named kRideRatingExcitementUndefined it would be clearer it's intended use is for excitement only I think - this way it was an easy assumption to make it can be used for the other two ratings when setNulling them all.